SMLHelper/Handlers/CraftTreeHandler.cs

```diff
@@ -81,86 +40,24 @@ public static ModCraftTreeRoot GetExistingTree(CraftTree.Type Scheme)
-        /// </summary>
-        /// <param name="craftTree">The target craft tree to edit.</param>
-        /// <param name="craftingItems">The item to craft.</param>
-        /// <param name="stepsToTab">
-        /// <para>The steps to the target tab.</para>
-        /// <para>These must match the id value of the CraftNode in the crafting tree you're targeting.</para>
-        /// <para>Do not include "root" in this path.</para>
-        /// </param>
-        /// <seealso cref="GetExistingTree"/>
-        /// <seealso cref="ModCraftTreeRoot.GetTabNode(string[])"/>
-        /// <seealso cref="ModCraftTreeLinkingNode.AddCraftingNode(IEnumerable{TechType})"/>
-        public static void AddCraftingNode(CraftTree.Type craftTree, IEnumerable<TechType> craftingItems, params string[] stepsToTab)
-        {            
-            ModCraftTreeRoot root = GetExistingTree(craftTree);
-
-            if (root == null)
-                return;
-
-            ModCraftTreeTab tab = root.GetTabNode(stepsToTab);
-
-            if (tab == null)
-                return;
-
-            tab.AddCraftingNode(craftingItems);
+            // Add to game
+            CraftTreePatcher.CraftingNodes.Add(new CraftingNode(stepsToTab, craftTree, craftingItem));
         }
 
         /// <summary>
         /// Adds a new crafting node to the root of the specified crafting tree
         /// </summary>
         /// <param name="craftTree">The target craft tree to edit.</param>
         /// <param name="craftingItem">The item to craft.</param>
-        /// <seealso cref="GetExistingTree"/>
         /// <seealso cref="ModCraftTreeRoot.GetTabNode(string[])"/>
         /// <seealso cref="ModCraftTreeLinkingNode.AddCraftingNode(TechType)"/>
         public static void AddCraftingNode(CraftTree.Type craftTree, TechType craftingItem)
         {
-            ModCraftTreeRoot root = GetExistingTree(craftTree);
-
-            if (root == null)
-                return;
-
-            root.AddCraftingNode(craftingItem);
-        }
-
-        /// <summary>
-        /// Adds a collection of new crafting nodes to the root of the specified crafting tree.
-        /// </summary>
-        /// <param name="craftTree">The target craft tree to edit.</param>
-        /// <param name="craftingItems">The item to craft.</param>
-        /// <seealso cref="GetExistingTree"/>
-        /// <seealso cref="ModCraftTreeRoot.GetTabNode(string[])"/>
-        /// <seealso cref="ModCraftTreeLinkingNode.AddCraftingNode(IEnumerable{TechType})"/>
-        public static void AddCraftingNode(CraftTree.Type craftTree, IEnumerable<TechType> craftingItems)
-        {
-            ModCraftTreeRoot root = GetExistingTree(craftTree);
-
-            if (root == null)
-                return;
-
-            root.AddCraftingNode(craftingItems);
+            CraftTreePatcher.CraftingNodes.Add(new CraftingNode(new string[0], craftTree, craftingItem));
```

Comment:
Just noticed: These methods are now only for use with standard CraftTree.Types.
There's a bit more work to do here.
1. Remove the remaining XML Documentation referencing methods no longer used.
2. Add an Assert line to check that the CraftTree.Type is one of the standard trees.
3. Update the XML Documentation to mark that restriction.

Reply:
Updated in fc5f0f8